controllers/components/dashboard/dashboard.go

```diff
@@ -25,8 +25,16 @@ func (s *componentHandler) GetName() string {
 	return componentsv1.DashboardComponentName
 }
 
-func (s *componentHandler) GetManagementState(dsc *dscv1.DataScienceCluster) operatorv1.ManagementState {
-	return dsc.Spec.Components.Dashboard.ManagementState
+func (s *componentHandler) GetManagementState(dsc *dscv1.DataScienceCluster) string {
+	if s == nil {
+		return "Removed"
+	}
+	switch dsc.Spec.Components.Dashboard.ManagementState {
+	case operatorv1.Managed, operatorv1.Removed:
+		return string(dsc.Spec.Components.Dashboard.ManagementState)
+	default: // Force and Unmanaged case for unknown values, we do not support these yet
+		return "Unknown"
```

Comment:
Wonder if this should return `operatorv1.Removed`, in other words, how do we generally handle the case where the `ManagementState` is not set ?

Reply:
i think it falls into the `s == nil `case ?

Reply:
Now, i think i understand your comments better:
i can think of two approaches:
1. remove this default for now: since enum is only for Removed and Managed. till we add support for Force and Unmanaged, we do not need to worry about "default"
2. change the whole funciton to
```
func (s *componentHandler) GetManagementState(dsc *dscv1.DataScienceCluster) operatorv1.ManagementState {
	if dsc.Spec.Components.Dashboard.ManagementState == operatorv1.Managed {
		return operatorv1.Managed
	}
	return operatorv1.Removed
}
```

Reply:
yep something like option 2

Reply:
updated